### ISSUE TITLE
Control Seals Vanish - randomoly

### ISSUE THREAD

**Issue by robbcrg** [Azanor/thaumcraft-beta#907]
I enter my multiple world after a day or so, no changes, and my golems are just idle and all control seals are gone. I cannot find any pattern. Server restarts do not seem to cause it.

Fixing the control seals was not a big deal when I had a simple setup, but as the setups become more complex and colors are used it's daunting to set everything up again.

One thing that might be related, when this happens my missing golems seem to have returned. I figured they just wander off and I could not find them. Now I am not so sure.

I have seen these 3 or 4 times now. 

I cannot reproduce this reliably and the only reference I found was going back to Thaumcraft 4 in 2016

https://www.reddit.com/r/allthemods/comments/41dkki/missing_seals/

**Comment by lilolbear** [comment]
I believe that I have run into this one in B24. It's a private multiplayer and all the seals I had placed, 8 in total, have vanished. I do not recall if I placed them prior to or after switching from B22 but they were functional in B24 at one point. I'll keep my eye out for any further evaporating seals so I can locate potential logs.

**Comment by Azanor** [comment]
Thanks, I have been unable to replicate this so any logs would be great.

**Comment by Amelia-** [comment]
I have noticed this happening a few times as well also on a private multiplayer server. It has been happening from I think beta 16? And still happens on beta 24.

I have not been able to find out what causes it. One thing I did find odd about this is that during all this time where all the other seals disappeared, a collect seal placed on the ground has not.

**Comment by Schmocky** [comment]
I observed the same behavior as Ameilia. At one point all seals just disappeared on our server and i just rebuild them. Been like 16 to 20 of them. 
After that i only build two farms with golems again up again and now only the Collect and Harvest seals vanished. The store seals however are still on the chests.

Sadly i can not provide any logs. It's beta 24 on FTB Revelations

**Comment by sniggyfigbat** [comment]
This still appears to be an outstanding issue - I and a few friends are running an Enigmatica 2 Expert (1.67) private server and golems are essentially unusable due to seal disappearance.

Is there anything we could do to help isolate and reproduce the bug? We'd be happy to send logs or a full copy of the server files, if that would be of use.

**Comment by Fennikk** [comment]
I also have been having the same issues with it, and it sucks seeing as I adore using golems for everything even when pipes would be sufficent.. I just really like seeing them *do* stuff.

But It breaks my heart every time I log on and my golem I use to guard an area or something can't do anything because of his seal just being.. gone.

**Comment by Chis09** [comment]
I too was having this issue in an Enigmatica 2 Expert 1.64 private server. After a bit of playing around I realized that the seals only disappeared if the block they were _placed_ on was not chunk loaded. It didn't always happen immediately (sometimes after unloading the chunks then reloading, the seals were still there) but the seals did always eventually disappear. After placing seals on blocks that were always chunk loaded I have not had any issues with disappearing seals. 

TL;DR it seems that the blocks the seals are _placed_ on must remain chunk loaded

**Comment by z2oh** [comment]
I got my seals to stop disappearing by keeping them chunk loaded, but the golems will stop working after ~15 minutes or so. I have to restart the server to get them doing anything again. I am using B26 as a part of Enigmatica 2 Expert 1.71.

**Comment by AppleSaws** [comment]
Just wanted to say that I just had this occur for me as well. Sorry for reviving the old post, but I figured it made more sense than making a new post. I'm playing the most recent update of Enigmatica 2: Expert on a friend's private multiplayer server if that helps at all.

To give more detail, I had a collect, harvest, and store seal placed, all 3 of which disappeared. I believe that I may have unloaded the chunk for a short bit between the last time I saw the seals present and when I noticed the farm was overgrown and the seals were missing. The harvest seal was in the center of the farm on top of the fertile soil block, the collect seal was on the bottom of said block, and the store seal was on a barrel on the edge of the farm on top of the border ring of barrier stones. The farm had no issues before the seals disappeared (outside of the golem not being able to reach the sugar cane that fell on the barrier stones, but that wasn't a big deal). Nothing else about the farm appeared to have changed.

Edit: I looked over at my farm again and saw it overgrown again. I went over and checked, and the seals were still there, but the golem was not doing anything. Changing the settings on the seals did nothing, and breaking and replacing the seals did not work either. Picking up and replacing the golem also had no effect. Picking up the golem, disconnecting, then reconnecting in and placing it didn't work, and neither did just disconnecting and reconnecting with the golem placed. Damaging the golem did not work either. I can't test restarting the server since I don't run the server, but it's possible that that could fix it, since it worked for z2oh. Unlike the previous event where the seals disappeared, the farm stayed loaded the entire time. I then picked up  my harvest seal (Just in cast it despawned. I left the store and collect seals), then went somewhere far away, stayed for a number of minutes, then returned. The seals were gone, and the golem was obviously not functioning due to the lack of deals. Breaking the blocks that either seal was connected to did not return either seal, and relogging did not help. However, placing down the harvest seal caused the golem to begin functioning again.

Overall, I would guess that unloading and then reloading the chunk fixes the golem problem, though removes any seals left there. It's also worth noting that in my original testing I never tried removing all 3 of my seals and then replacing them, though I'm not sure why that would have re-activated the golem.

All of this testing was done on Enigmatica 2 Expert, version 1.82a.

One last edit: So, I played some more, visited the end for a bit, came back. The control seals were still there, but the golem had stopped working, though it was holding a piece of sugar cane, presumably from before it had stopped working. I was able to pick up the golem no problem and re-place it, but it had no effect. Furthermore, I was completely unable to pick up or modify any of the control seals I had placed with the golemancer's bell, despite being able to see them when holding the bell or similar. Attempting to break the blocks the seals were attached to also failed to work. I logged out and then back in and the golem started working again (I don't remember exactly what I did to the golem, I think I had it in my inventory when I logged out and then placed it again?), responding to golemancer bell commands. The old sigils were still visible, but the golem ignored them. I crafted one more sigil and placed it, and the golem responded to it. I was also able to place the new sigil on top of where the old, non-working sigils appeared to be, seemingly ending what was probably a graphical glitch of the old sigils appearing to be there while not actually being there.

I'm done with golems and control sigils. They just don't work.
